orchagent/portsorch.cpp

```diff
@@ -3867,9 +3878,9 @@ void PortsOrch::doPortTask(Consumer &consumer)
                 if (pCfg.fec.is_set)
                 {
                     /* reset fec mode upon mode change */
-                    if (!p.m_fec_cfg || p.m_fec_mode != pCfg.fec.value)
+                    if (!p.m_fec_cfg || p.m_fec_mode != pCfg.fec.value || p.m_override_fec != pCfg.fec.override_fec)
                     {
-                        if (!isFecModeSupported(p, pCfg.fec.value))
+                        if (!isFecModeSupported(p, pCfg.fec.value, pCfg.fec.override_fec))
```

Comment:
@dgsudharsan can we keep the existing function not coupled with fec override and indepently taken care outised?

Reply:
Done